### PR TITLE
Tooling phase 2

### PR DESCRIPTION
**Pull request by lauckhart**
This migrates remaining packages to new tooling.

Includes a number of cosmetic and quality-of-life upgrades.

Fixes numerous issues exposed or created by new toolchain.

Key bits:
- Migrated all modules to standard base tsconfig
- Migrated all modules to Mocha
- Add various settings to build & test commands
- Added "run" command that behaves similar to ts-node
- Moved much of the mocking from individual tests into centralized locations
- In particular, significantly reworked time management in tests
- Rewrote matter-node.js registration (old way wasn't compatible w/ ESM semantics)
- Convert Shell.js to typescript (new config requires type declarations)
- Move chip testing to chip-testing/ under root
- Completely remove Jest and ts-node dependencies
- Fixed declaration source map paths
- Added logic to hide successful logs in test setup/cleanup
- Migrated matter-node.js setup from module scope to before hooks
- Added utility file to support mocharc.cjs implementations
- Filter noise from Node assert errors in testing

### CODE REVIEW

File: packages/matter.js/src/crypto/Crypto.ts [hunk 1]
```diff
@@ -123,3 +123,7 @@ export abstract class Crypto {
 // place.
 import elliptic from "elliptic";
 export const { ec } = elliptic;
+
+if (typeof MatterHooks !== "undefined") {
```
Comment (Apollon77) [inline packages/matter.js/src/crypto/Crypto.ts:127]: Maybe add a comment here what this is ... I did not got that till the end ...

File: packages/matter.js/src/log/Logger.ts [hunk 2]
```diff
@@ -346,14 +346,6 @@ export class Logger {
...
-    /**
-     * This global hook allows test suites to hook logging independent of the
-     * module mechanism.
-     */
-    let MatterLoggerHook: ((TheLogger: typeof Logger) => void) | undefined;
-}
-
-if (typeof MatterLoggerHook !== "undefined") {
-    MatterLoggerHook(Logger);
+if (typeof MatterHooks !== "undefined") {
```
Comment (Apollon77) [inline packages/matter.js/src/log/Logger.ts:349]: same here

File: packages/matter.js/src/time/Time.ts [hunk 3]
```diff
@@ -59,3 +59,7 @@ export interface Timer {
     /** Stops this timer, chainable. */
     stop(): Timer;
 }
+
+if (typeof MatterHooks !== "undefined") {
```
Comment (Apollon77) [inline packages/matter.js/src/time/Time.ts:63]: here

File: packages/matter.js/test/cluster/definitions/WindowCoveringClusterTest.ts [hunk 4]
```diff
@@ -4,21 +4,29 @@
...
-    }) as Crypto;
-
 import { WindowCovering, WindowCoveringCluster } from "../../../src/cluster/definitions/WindowCoveringCluster.js";
 import { ClusterServer } from "../../../src/cluster/server/ClusterServer.js";
+import { Crypto } from "../../../src/crypto/Crypto.js";
 import { BitFlags } from "../../../src/schema/BitmapSchema.js";
 
 describe("WindowCoveringCluster", () => {
     const WindowCovering_LF_PALF = WindowCoveringCluster.with("Lift", "PositionAwareLift");
+    let actualCrypto = Crypto.get;
```
Comment (Apollon77) [inline packages/matter.js/test/cluster/definitions/WindowCoveringClusterTest.ts:14]: Why this needs to move here? Is not enoigh mocked?

Comment (lauckhart) [reply packages/matter.js/test/cluster/definitions/WindowCoveringClusterTest.ts:14]: Left in accidentally, I think I moved it here initially before I realized I needed to edit a bunch of files and moved it to the test harness

File: packages/matter-node.js/package.json [hunk 5]
```diff
@@ -53,160 +53,218 @@
...
+                "types": "./dist/esm/exports/tlv.d.ts",
+                "default": "./dist/esm/exports/tlv.js"
+            }
         },
         "./*": {
-            "types": "./dist/cjs/*/export.d.ts",
-            "default": "./dist/cjs/*/export.js"
-        }
-    },
-    "typesVersions": {
```
Comment (Apollon77) [inline packages/matter-node.js/package.json:154]: This needs to stay because if is neededf for backward compatibility issues.

Comment (Apollon77) [reply packages/matter-node.js/package.json:154]: I readd

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**